tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,696 @@
+    /// The values that we wanna match on, (true, false, maybe)
+    static ref PII_VALUES: Mutex<Vec<String>> = Mutex::new(vec![]);
+    /// Maps the name of the module to all of the use statements therein, plus the path of the
+    /// types that are defined there.
+    static ref USE_STATEMENTS: Mutex<BTreeMap<String, BTreeSet<String>>> = Mutex::new(BTreeMap::new());
+}
+
+/// It's only purpose is to populate ALL_TYPES and USE_STATEMENTS
+struct FileSyntaxVisitor {
+    module_path: String,
+}
+
+impl FileSyntaxVisitor {
+    fn insert_use_statements(&self, use_statements: Vec<String>) {
+        USE_STATEMENTS
+            .lock()
+            .unwrap()
+            .entry(self.module_path.clone())
+            .or_default()
+            .extend(use_statements);
+    }
+
+    fn fill_pii_hashmaps(paths: &Vec<PathBuf>) {
+        for path in paths {
+            // first iterate through all rust files to create the hashmap
+            let mut hashmap_filler = FileSyntaxVisitor {
+                module_path: rust_file_to_use_path(path.as_path()),
+            };
+            let file_content = fs::read_to_string(path.as_path()).unwrap();
+            let syntax_tree: syn::File = syn::parse_file(&file_content).unwrap();
+            hashmap_filler.visit_file(&syntax_tree);
+        }
+    }
+}
+
+impl<'ast> Visit<'ast> for FileSyntaxVisitor {
+    fn visit_item_struct(&mut self, node: &'ast ItemStruct) {
+        // These are pretty common and we know they dont contain PII so might as well not include
+        // them to save time.
+        for common_value in ["Annotated", "Value"] {
+            if node.ident == common_value {
+                return;
+            }
+        }
+
+        let struct_name = format!("{}::{}", self.module_path, node.ident);
+        self.insert_use_statements(vec![struct_name.clone()]);
+        ALL_TYPES
+            .lock()
+            .unwrap()
+            .insert(struct_name, EnumOrStruct::Struct(node.clone()));
+    }
+
+    fn visit_item_enum(&mut self, node: &'ast ItemEnum) {
+        let enum_name = format!("{}::{}", self.module_path, node.ident);
+        self.insert_use_statements(vec![enum_name.clone()]);
+        ALL_TYPES
+            .lock()
+            .unwrap()
+            .insert(enum_name, EnumOrStruct::Enum(node.clone()));
+    }
+
+    fn visit_item_use(&mut self, i: &'ast syn::ItemUse) {
+        let use_statements = usetree_to_paths(&i.tree, &self.module_path)
+            .iter()
+            .filter(|s| s.contains("relay"))
+            .cloned()
+            .collect();
+
+        self.insert_use_statements(use_statements);
+    }
+}
+
+#[derive(Ord, PartialOrd, PartialEq, Eq, Hash, Clone, Debug, Default)]
+struct TypeAndField {
+    // Full path of a type. E.g. relay_common::protocol::Event, rather than just 'Event'.
+    qualified_type_name: String,
+    field_name: String,
+}
+
+fn find_rs_files(dir: &PathBuf) -> Vec<std::path::PathBuf> {
```

Comment:
In `fill_pii_hashmaps` we use `Path`s instead of `PathBuf`s. Could we change the return type and make the following methods easier?